test/models/changeset/issue_comment_test.rb

```diff
@@ -29,4 +29,33 @@
       Changeset::IssueComment.valid_webhook?(webhook_data).must_equal false
     end
   end
+
+  describe '#sha' do
+    let(:issue_data) { {issue: {number: 1}}.with_indifferent_access }
+
+    let(:pr_data) do
+      {
+        head: {
+          sha: 'abcd123',
+          ref: 'a/test'
+        }
+      }.with_indifferent_access
+    end
+
+    let(:cached_pr_data) do
+      {
+        head: {
+          sha: 'cach123',
+          ref: 'a/test'
+        }
+      }.with_indifferent_access
+    end
+
+    it 'does not read from cached PullRequest' do
```

Comment:
giving a reason here would also be helpful for future archeologists

`... does not read from cache to avoid foo bar` ... ideally a higher level test like `shows new comment when user updates pull request` ... so it fails with an error that makes sense / reproduces the actual issue